### DISCUSSION TITLE
Include gaps from i3-gaps to main i3

### DISCUSSION THREAD

**Discussion by pietryszak** [i3/i3#5237] [Ideas]
How long we need to wait to implement or merge at one project, i3-gaps to i3. 
I see that maintainer of i3-gaps has plans to do that, but in 2022 still i3 is without possibility to use gaps.
Is total mess when on one distro is i3 with gaps and in another distro is only i3.

**Comment by Airblader** [comment]
You're welcome to contribute to #3724 yourself or hire someone to do it for you if you can no longer wait for someone else to do it in their free time.

**Reply by pietryszak** [reply]
It was a question, in meaning someone has really plan to do that or just plans that never happens. Any date? Next year, 2024 or no data when this happens at all?

**Reply by Airblader** [reply]
The way open source works is that you cannot tell people when they need to do something.

We did recently have another discussion among maintainers and I am going to post that in #3724 in a minute. But there won't be any timeline because this is work people do for free in their personal time.

**Reply by pietryszak** [reply]
Ok. So i stay with i3-gaps till gaps was included in main i3. Thank you for information ☺️

**Reply by ghost** [reply]
If you're on a rolling release distro like arch, both variants officially exist btw. It's also very much possible that gaps going to be merged into the official i3 someday soon.

**Reply by okraits** [reply]
It already happened: https://github.com/i3/i3/pull/5241

**Reply by pietryszak** [reply]
> If you're on a rolling release distro like arch, both variants officially exist btw. It's also very much possible that gaps going to be merged into the official i3 someday soon.

I don't use arch btw. Ubuntu + i3-gaps ;)

**Reply by ghost** [reply]
@Airblader  Amazing. Btw is this got changed [smart_borders -> hide_edge_borders](https://github.com/i3/i3/blob/next/docs/userguide#L788) ?

**Reply by pietryszak** [reply]
> @Airblader Amazing. Btw is this got changed [smart_borders -> hide_edge_borders](https://github.com/i3/i3/blob/next/docs/userguide#L788) ?

If a smart_borders change name, a lot of config files of people needs to be updated

**Reply by ghost** [reply]
Yah. It seems that's the case.

**Reply by Airblader** [reply]
Are you referring to `hide_edge_borders smart_no_gaps`? That has been implemented, but it seems it may have been forgotten to be added to the docs. The "smart" option is different, and has existed previously already.

**Reply by ghost** [reply]
I've never knew `hide_edge_borders` even exist. My config has `smart_borders on` from the beginning (including smart_gaps).

**Reply by Airblader** [reply]
The docs for smart_borders were also overlooked. The feature has been implemented, though, no worries. I'll make sure we fix the docs. Thanks for the pointers.

**Reply by pietryszak** [reply]
> I've never knew `hide_edge_borders` even exist. My config has `smart_borders on` from the beginning (including smart_gaps).

Same, I use a `smart_borders on` but never know that  `hide_edge_borders`  exist.
@Airblader Good to see a merge starts ;)
